pkg/autoscaler/autoscaler/tikv_autoscaler.go

```diff
@@ -35,8 +38,12 @@ func (am *autoScalerManager) syncTiKV(tc *v1alpha1.TidbCluster, tac *v1alpha1.Ti
 		emptyAutoScalingCountAnn(tac, v1alpha1.TiKVMemberType)
 		return nil
 	}
-	currentReplicas := getStateUpReplicas(tc)
-	targetReplicas := calculateRecommendedReplicas(tac, v1alpha1.TiKVMemberType, client)
+	instances := filterTiKVInstances(tc)
+	currentReplicas := int32(len(instances))
+	targetReplicas, err := calculateTikvMetrics(tac, sts, client, instances)
```

Comment:
This is the entrance function for TiKV to calculate auto-scaling by each metric.